Source/Core/Core/PowerPC/Jit64/Jit_Branch.cpp

```diff
@@ -204,6 +290,12 @@ void Jit64::bcctrx(UGeckoInstruction inst)
     if (inst.LK_3)
       MOV(32, PPCSTATE_LR, Imm32(js.compilerPC + 4));  // LR = PC + 4;
     AND(32, R(RSCRATCH), Imm32(0xFFFFFFFC));
+    if (IsDebuggingEnabled())
+    {
+      const BitSet32 caller_save = CallerSavedRegistersInUse() | BitSet32{RSCRATCH};
+      const RCX64Reg bw_reg_b = gpr.Scratch();
```

Comment:
I suspected what you have said about `RegCache::Scratch()` recycling a register that is caller saved, but figured it was an unlikely missed optimization and not inherently unsafe. However, your advice for what is safe after a GPR flush is something I had not figured out, so thank you.

Reply:
Technically it isn't unsafe – you're just pushing a register that you don't necessarily need to push – but it is a bug, and we want the JIT to be as performant as possible.

RSCRATCH_EXTRA is a register you can normally only use if you asked the register cache to reserve it for you, but since you've flushed all registers, you can just go ahead and use it without asking the register cache.